public/app/features/dashboard-scene/scene/ReloadOnFiltersChangeBehavior.ts

```diff
@@ -0,0 +1,38 @@
+import { SceneObjectBase, SceneObjectState, VariableDependencyConfig } from '@grafana/scenes';
+import { appEvents } from 'app/core/core';
+import { ReloadDashboardEvent } from 'app/types/events';
+
+export interface ReloadOnFiltersChangeBehaviorState extends SceneObjectState {
+  reloadOnFiltersChange?: string[] | boolean;
+  uid?: string;
+}
+
+export class ReloadOnFiltersChangeBehavior extends SceneObjectBase<ReloadOnFiltersChangeBehaviorState> {
+  public constructor(state: ReloadOnFiltersChangeBehaviorState) {
+    super(state);
+
+    if (state.reloadOnFiltersChange && state.uid) {
+      this._variableDependency = new VariableDependencyConfig(
+        this,
+        state.reloadOnFiltersChange === true
+          ? {
+              onAnyVariableChanged: this.reloadDashboard.bind(this),
+            }
+          : {
+              variableNames: state.reloadOnFiltersChange,
+              onVariableUpdateCompleted: this.reloadDashboard.bind(this),
+            }
```

Comment:
Don't understand this. So when the `reloadOnFiltersChange` is true it will reload on any variable changed, but otherwise on specified filters? The naming here is quite confusiong. Also - don't like `reloadOnFiltersChange` is either string array or boolean. It's tricky to reason about and handle. Do we even need this to be typed as `string[]`?

Reply:
Yeah. There is the need to reload on specific variables or on all variables. Additionally, after the latest discussions there is also the need to reload on specific labels for adhoc filters.

I'll add multiple different metas for those behaviors:
- `reloadOnAnyVariableChange`: `true` / `false` - it'll reload on any change. This will be the primary check.
- `reloadOnVariablesChange`: `string[]` - it'll reload on variables with the names specified in the array. If an adhoc variable is specified, it will reload on any label change for the specific adhoc variable.
- `reloadOnAdhocLabelsChange`: `string[]` - it'll reload on specific labels in adhoc filters, regardless of the name of the adhoc variable. If a variable is specified in `reloadOnVariablesChange` but specific labels will be specified in `reloadOnAdhocLabelsChange`, the mechanism will ignore the latter.

Reply:
See my comment https://raintank-corp.slack.com/archives/C068ZP1HBGQ/p1728898992409929?thread_ts=1727953353.962049&cid=C068ZP1HBGQ, maybe we can move it here to have the full discussion